ADIOS2_USE_Endian_Reverse Broken

@eschnett reported in Spack that `-DADIOS2_USE_Endian_Reverse=ON` is broken: https://github.com/spack/spack/issues/23643
```
adios2/helper/adiosMemory.cpp:32:22: error: no match for 'operator==' (operand types are 'const adios2::DataType' and 'const char [1]')
   32 |         if (destType == "")
      |             ~~~~~~~~ ^~ ~~
      |             |           |
      |             |           const char [1]
      |             const adios2::DataType
```

I can reproduce this with the latest `master` as of d6547823360fe721ac065d731af4404ff07ff174

```
cmake -S . -B build -DADIOS2_USE_Endian_Reverse=ON
cmake --build build
```

cc @pnorbert @chuckatkins

--- a/source/adios2/helper/adiosMemory.cpp
+++ b/source/adios2/helper/adiosMemory.cpp
@@ -29,7 +29,7 @@ void CopyPayloadStride(const char *src, const size_t payloadStride, char *dest,
 #ifdef ADIOS2_HAVE_ENDIAN_REVERSE
     if (endianReverse)
     {
-        if (destType == "")
+        if (destType == DataType::None)
         {
         }
 #define declare_type(T)                                                        \

